### ISSUE TITLE
Streaming to Youtube not successful

### ISSUE THREAD

**Issue by wepe53** [manishkatyan/bbb-streaming#6]
Hi all,
I tried to implement streaming from BBB (V2.4-rc-4 running on MS Azure) to a youtube channel but failed.
I also tried more recent BBB V2.4.6 on my private server but also failed!

Everything looks fine in the beginning: a user named "live" is shown in the conference but no data is transferred to the Youtube channel. After a while user live is offline and finally  disappears.

I am pretty sure that all parameters  (especially Youtube URL and key) are fine because I tested streaming with OBS studio without any problems. Also Wireshark does not show any RTMP record or IP/UDP-record for target "[a.rtmp.youtube.com](http://a.rtmp.youtube.com/)"

I wonder whether there is a problem in the app i.e. for whatever reason streaming to Youtube does not take place at all - but why?

**Unfortunately there is no log or debug option.** Also I am not  familiar with Docker.

Any idea what may cause the problem?

Thanks

Wolfgang

**Comment by arun-hel** [comment]
@wepe53  We are working on it.

**Comment by wepe53** [comment]
One more comment/question:
I also tried to run bbb-streaming on a **remote** Ubuntu system (not the BBB server) - is that possible at all?
The script did not produce any error but I recognized that no user "live" was created in BBB conference.
So maybe you should state that bbb-streaming **must** run on the BBB platform itself??

**Comment by wepe53** [comment]
I tried to gather some log information (with the help of Google) - maybe this helps you to analyze the problem:

conveyAdmin@conference2:~/youtube/bbb-streaming$ sudo docker-compose up -d
Creating network "bbb-streaming_default" with the default driver
Creating liveStreaming ... done
conveyAdmin@conference2:~/youtube/bbb-streaming$ sudo docker container logs liveStreaming
INFO:root:Starting browser!!
INFO:root:streaming intro...
ffmpeg version 4.3.2-0york0~18.04 Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.5.0-3ubuntu1~18.04)
  configuration: --prefix=/usr --extra-version='0york0~18.04' --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libzimg --enable-pocketsphinx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
false: No such file or directory
INFO:root:Open BBB and hide elements!!
INFO:root:updated get_join_url...
INFO:root:https://conference2.germanywestcentral.cloudapp.azure.com/bigbluebutton/api/join?meetingID=test1&fullName=Live&password=ap&userdata-bbb_auto_join_audio=true&userdata-bbb_enable_video=true&userdata-bbb_listen_only_mode=true&userdata-bbb_force_listen_only=true&userdata-bbb_skip_check_audio=true&joinViaHtml5=true&userdata-bbb_auto_swap_layout=true&checksum=a0d241fed2d6e384f34db298b50287940054f654
Traceback (most recent call last):
  File "stream.py", line 212, in <module>
    bbb_browser()
  File "stream.py", line 120, in bbb_browser
    browser.execute_script("document.getElementById('container').setAttribute('style','margin-bottom:30px');")
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriver.py", line 636, in execute_script
    'args': converted_args})['value']
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.JavascriptException: Message: javascript error: Cannot read property 'setAttribute' of null
  (Session info: chrome=91.0.4472.77)

conveyAdmin@conference2:~/youtube/bbb-streaming$

**Comment by arun-hel** [comment]
We updated the bbb-streaming to support 2.4 (successfully tested on BBB 2.4.6). Check it out.

For Installtion refer https://github.com/manishkatyan/bbb-streaming#install

**Comment by wepe53** [comment]
Great! I will give it a try later tonight!
Thanks!

Arunkumar Bhat ***@***.***> schrieb am Do., 5. Mai 2022,
09:27:

**Comment by wepe53** [comment]
Hi,
I got it working - thanks a lot! I need more time for testing!

So far I made 1 significant observation: there is a delay of about 1 to 2 minutes between live BBB conference (which runs on a virtual MS Azure server) and Youtube stream - what/who is causing this?

I did not see such a delay when testing streaming with OBS (however OBS did run on my Windows 11 PC and I streamed a Chrome tab with a BBB conference window) !!!

:-) Wolfgang
